Source/Details/ASDataController.mm

```diff
@@ -437,6 +437,29 @@ - (void)waitUntilAllUpdatesAreCommitted
   [self _scheduleBlockOnMainSerialQueue:^{ }];
 }
 
+- (BOOL)isProcessingUpdates
+{
+  ASDisplayNodeAssertMainThread();
+  long waitResult = dispatch_group_wait(_editingTransactionGroup, DISPATCH_TIME_NOW);
+  return (waitResult == 0 ? NO : YES);
```

Comment:
Yeah, this is a good catch. I think this might be possible.

If we flush the main serial queue here, you could get situations where calling isProcessingUpdates results in calls out to the delegate / dataSource, which would be pretty weird and maybe unexpected by clients. So I think I'll just check the length of the queue.